### PR TITLE
feat: recent session history

### PR DESCRIPTION
**Pull request by yomybaby**
This pull request introduces several changes to the `react` project to enhance the session management features and improve form handling. The most important changes include adding a new `SessionTemplateModal` component, and enhancing the `SessionLauncherPage` with session history functionality.

### New Features and Components:
* **`SessionTemplateModal` Component**: Added a new modal component to display recent session history, allowing users to apply previous session settings easily. (`react/src/components/SessionTemplateModal.tsx`)

### Updates to Session Management:
* **Session History Handling**: Introduced a new hook `useRecentSessionHistory` to manage recent session history, including functions to push, clear, and remove session history entries. (`react/src/hooks/backendai.tsx`)
* **`SessionLauncherPage`**: Enhanced to include session history functionality, allowing users to start sessions with previously used settings and view recent session history. (`react/src/pages/SessionLauncherPage.tsx`) [[1]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1R229-R239) [[2]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1L542-R575) [[3]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1L652-R694)

### Minor Changes:
* **Code Cleanup and Imports**: Various files updated to clean up imports and improve code readability. (`react/src/pages/SessionLauncherPage.tsx`, `react/src/hooks/useBAISetting.tsx`, `react/src/hooks/backendai.tsx`) [[1]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1R34) [[2]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1L48-R52) [[3]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1L62-R66) [[4]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1L72) [[5]](diffhunk://#diff-cb5aa3700f469823d540634a1acf2796ffcfa96416d6fa9333bfab0ff658b2c1L95-R98)

These changes collectively enhance the user experience by providing better session management tools and improving the flexibility and functionality of existing components.

### CODE REVIEW

**Comment by yomybaby** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2742?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2742** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2742?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2741** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2741?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2740](https://github.com/lablup/backend.ai-webui/pull/2740) <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2740?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yomybaby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/VFolderTableFormItem.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const VFolderTableFormItem: React.FC<VFolderTableFormItemProps> = ({
 }) => {
   const form = Form.useFormInstance();
   const { t } = useTranslation();
-  Form.useWatch('vfoldersAliasMap', form);
+  console.log(Form.useWatch('vfoldersAliasMap', form));
```
Comment (agatha197) [inline react/src/components/VFolderTableFormItem.tsx]:
```suggestion
```

File: react/src/pages/SessionLauncherPage.tsx [hunk 2]
```diff
@@ -290,7 +296,13 @@ const SessionLauncherPage = () => {
   }, []);
 
   const mergedInitialValues = useMemo(() => {
-    return _.merge({}, INITIAL_FORM_VALUES, formValuesFromQueryParams);
+    return _.merge(
+      {},
+      // bec
+      // { sessionName: '' } as Partial<SessionLauncherFormValue>,
```
Comment (agatha197) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion
```

File: react/src/pages/SessionLauncherPage.tsx [hunk 3]
```diff
@@ -539,58 +552,72 @@ const SessionLauncherPage = () => {
...
+              sessionId: string;
+              sessionName: string;
+              servicePorts: Array<{ name: string }>;
+            }>) => {
+              pushSessionHistory({
+                id: firstSession.sessionId,
+                params: usedSearchParams,
+                name: firstSession.sessionName,
+              });
+              // console.log('##sessionPromises', firstSession);
```
Comment (agatha197) [inline react/src/pages/SessionLauncherPage.tsx]:
```suggestion
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
Works fine. I just left minor comment to remove debugging codes.

**Comment by yomybaby** [comment]
Thank you for reviewing. I have accepted your suggestion.

**Review by lizable — APPROVED** [review]
LGTM.

**Review by agatha197 — APPROVED** [review]
LGTM

**Pull request merged**